Testscripts/Linux/ETHTOOL-CHECK-CHANNEL.sh

```diff
@@ -52,42 +51,34 @@ fi
 
 # Get number of channels
 channels=$(ethtool -l "${SYNTH_NET_INTERFACES[@]}" | grep "Combined" | sed -n 2p | grep -o '[0-9]*')
+# Get max number of channels
+max_channels=$(ethtool -l "${SYNTH_NET_INTERFACES[@]}" | grep "Combined" | sed -n 1p | grep -o '[0-9]*')
 # Get number of cores
 cores=$(grep -c processor < /proc/cpuinfo)
 
-if [ "$channels" != "$cores" ]; then
-    LogErr "Expected: $cores channels and actual $channels."
-    SetTestStateFailed
-    exit 0
-fi
-
-LogMsg "Number of channels: $channels and number of cores: $cores."
-
-let new_channels=cores-1
-if [ $new_channels == 0 ]; then
-    LogErr "The number of cores should be greater than 1"
-    SetTestStateFailed
-    exit 0
-fi
+LogMsg "Number of channels: $channels, max number of channels: $max_channels and number of cores: $cores."
 
-# Change the number of channels
-sts=$(ethtool -L "${SYNTH_NET_INTERFACES[@]}" combined $new_channels 2>&1)
-if [[ "$sts" = *"Operation not supported"* ]]; then
-    LogErr "$sts"
-    LogErr "Change the number of channels of ${SYNTH_NET_INTERFACES[@]} with $new_channels channels failed"
-    SetTestStateFailed
-    exit 0
-fi
-
-# Get number of channels
-channels=$(ethtool -l "${SYNTH_NET_INTERFACES[@]}" | grep "Combined" | sed -n 2p | grep -o '[0-9]*')
-if [ "$new_channels" != "$channels" ]; then
-    LogErr "Expected: $new_channels channels and actual $channels after change the channels."
-    SetTestStateFailed
-    exit 0
-fi
+# Change number of channels with all values supported
+for new_channels in $(seq 1 $max_channels); do
+    old_channels=$channels
+    # Change the number of channels
+    sts=$(ethtool -L "${SYNTH_NET_INTERFACES[@]}" combined $new_channels 2>&1)
+    if [[ "$sts" = *"Operation not supported"* ]]; then
```

Comment:
ethtool might fail with other error, and you will be passing the test. check exit code too.

Reply:
yes but the channel is read back after the change and if it did not change will not pass the test